TELCODOCS-1050: Configuring the hub cluster to use unauthenticated registries (#55352)
openshift/openshift-docs: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc

:_content-type: PROCEDURE
[id="ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_{context}"]
= Configuring the hub cluster to use unauthenticated registries

You can configure the hub cluster to use unauthenticated registries.

> 

Author: Not that I'm aware of. @trewest, perhaps you could confirm if there are any other prereqs (network)?

> 

Reviewer: No I don't think any other prerequisites should be included. The cluster must be able to reach the registry over the network but I think that is common knowledge when working with registries.

.Procedure

> . Edit the `AgentServiceConfig` CR to include the `unauthenticatedRegistries` parameter:
> 
> [source,terminal]
> ----
> $ apiVersion: agent-install.openshift.io/v1beta1
> kind: AgentServiceConfig
> metadata:
>   name: agent
> spec:
>   unauthenticatedRegistries:
>   - example.registry.com
>   - example.registry2.com
>   ...
> ----

Reviewer 2: It might be a good idea to match the format of this procedure https://55352--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.html#enabling-assisted-installer-service-on-bare-metal_ztp-preparing-the-hub-cluster

```suggestion
. Update the AgentServiceConfig CR by running the following command:

[source,terminal]
----
$ oc edit AgentServiceConfig
---

. Add the `unauthenticatedRegistries` field in the CR:

[source,terminal]
----
apiVersion: agent-install.openshift.io/v1beta1
kind: AgentServiceConfig
metadata:
  name: agent
spec:
  unauthenticatedRegistries:
  - example.registry.com
  - example.registry2.com
  ...
----
```

Reviewer 3: +1 please make this change

Author: Updated as suggested.

Unauthenticated registries are listed under `spec.unauthenticatedRegistries` in the `AgentServiceConfig` resource.

> Any registry on this list is not required to have an entry in the pull secret validated by the `assisted-service`.

Reviewer: I would clarify that it is the pull secret created for the spoke cluster

Author: Updated as suggested.

> `assisted-service` validates the [pull secret](hive-integration/kube-api-getting-started.md#2-create-a-pull-secret) provided for spoke cluster installation by making sure it contains the authentication information for every image registry used for installation.

Author: Updated as suggested.

[NOTE]
====
[Mirror registries](#mirror-registry-configuration) will automatically be added to the ignore list and do not need to be added under `spec.unauthenticatedRegistries`.

> Specifying the `PUBLIC_CONTAINER_REGISTRIES` environment variable in the [ConfigMap override](#specifying-environmental-variables-via-configmap) is still supported and will completely overwrite the list to whatever is in the override.

Reviewer 2: Likewise for these lines, the `[Mirror registries](#mirror-registry-configuration)` and `[ConfigMap override](#specifying-environmental-variables-via-configmap)` are leftover from a different document. Perhaps edit the linkage or remove the link altogether.

Author: Updated as suggested.

:_content-type: PROCEDURE
[id="ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_{context}"]
= Configuring the hub cluster to use unauthenticated registries

Use the following procedure to configure the hub cluster to use unauthenticated registries.

.Prerequisites

. Install <https://github.com/openshift/hive/blob/master/docs/install.md>[Hive]
. Install the <https://github.com/openshift/hive/blob/master/docs/using-hive.md#cluster-provisioning>[ClusterDeployment Custom Resource Definition (CRD)]
. Install <https://sdk.operatorframework.io/docs/installation>[operator-sdk]
. Install <https://github.com/kubernetes-sigs/kustomize/releases>[kustomize]
. Install <https://github.com/operator-framework/operator-registry/releases>[opm]

.Procedure

> . Build the operator bundle (optional)

Reviewer 2: Sorry, where in the official docs will this be located? The reason I ask is because the instructions for this procedure below directly refers to the assisted-service github repo

Author: Hi, @CrystalChun. The content is going in Scalability and Performance after [Configuring the hub cluster to use a disconnected mirror registry](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.html#ztp-configuring-the-cluster-for-a-disconnected-environment_ztp-preparing-the-hub-cluster) as part of 4.12.z. Thanks!

Registries defined in the `registries.conf` must be scoped by repository and not by registry.

[NOTE]
====
`quay.io/edge-infrastructure` and `mirror1.registry.corp.com:5000/edge-infrastructure` are both scoped by the `edge-infrastructure` repository. This is a valid configuration.

Removing the repository `edge-infrastructure` from location is an invalid configuration, as it will not pass openshift-installer validation.
====

  . Set the `mirrorRegistryRef` in the spec of `AgentServiceConfig` to the name of uploaded `ConfigMap` as follows:

[source,terminal]
----
$ cat <<EOF | kubectl apply -f -
apiVersion: agent-install.openshift.io/v1beta1
kind: AgentServiceConfig
metadata:
  name: agent
spec:
  databaseStorage:
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 10Gi
  filesystemStorage:
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 20Gi
  imageStorage:
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 10Gi
  mirrorRegistryRef:
    name: mirror-registry-config-map
EOF
----

> For more details on how to specify the CR, see [AgentServiceConfig CRD](https://github.com/openshift/assisted-service/blob/master/config/crd/bases/agent-install.openshift.io_agentserviceconfigs.yaml).

Reviewer 4: To generate a link here, you have to use the following syntax:
`link:<https://example.com>[Example Title]`

(Btw, I think I see the same problem on [line 179](https://github.com/openshift/openshift-docs/pull/55352/files#diff-409f6af30b90a6d04d6ba57f8de6ffe514553beb7f7c14717e92bd09dcc614e5R179))

Author: Updated as suggested.

:_content-type: PROCEDURE
[id="ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_{context}"]
= Configuring the hub cluster to use unauthenticated registries

> :context: ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries

Reviewer 3: Lose this! Only set `:context:` in assemblies :)

Author: Updated as suggested.

// Module included in the following assemblies:
//
// * scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc

:_content-type: PROCEDURE
[id="ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_{context}"]
= Configuring the hub cluster to use unauthenticated registries

You can configure the hub cluster to use unauthenticated registries.
Unauthenticated registries does not require authentication to access and download images.

.Prerequisites

* You have installed and configured a hub cluster and installed {rh-rhacm-first} on the hub cluster.

* You have installed the OpenShift Container Platform CLI (oc).

> * You have logged in as a user with cluster-admin privileges.

Reviewer 5: `cluster-admin` in backticks

Author: Updated as suggested.

* You have configured an unauthenticated registry for use with the hub cluster.

.Procedure

. Update the `AgentServiceConfig` custom resource (CR) by running the following command:
+
[source,terminal]
----

> $ oc edit AgentServiceConfig

Reviewer: This should technically be 
```
oc edit AgentServiceConfig agent
```

Author: Updated as suggested.

> ---

Reviewer 3: ```suggestion
----
```
Top and bottom codeblock delimiters need to match otherwise the build get's messed up.

Author: Updated as suggested.

. Add the `unauthenticatedRegistries` field in the CR:
+

> [source,terminal]

Reviewer 3: ```suggestion
[source,yaml]
```

Author: Updated as suggested.

----
apiVersion: agent-install.openshift.io/v1beta1
kind: AgentServiceConfig
metadata:
  name: agent
spec:
  unauthenticatedRegistries:
  - example.registry.com
  - example.registry2.com
  ...
----
+
Unauthenticated registries are listed under `spec.unauthenticatedRegistries` in the `AgentServiceConfig` resource.
Any registry on this list is not required to have an entry in the pull secret validated by the `assisted-service`.

`assisted-service` validates the pull secret created for the spoke cluster installation by making sure it contains the authentication information for every image registry used for installation.

[NOTE]
====
Mirror registries are automatically added to the ignore list and do not need to be added under `spec.unauthenticatedRegistries`.

> 

Reviewer 2: > @CrystalChun should we include a comment with `PUBLIC_CONTAINER_REGISTRIES` defaults (i.e. quay.io, registry.svc.ci.openshift.org)?

Agreed, @ktothill please include these defaults. I think here would be best, wdyt @ktothill @trewest?

Author: I can add those defaults. Are there any others outside of [quay.io](http://quay.io/), [registry.svc.ci.openshift.org](http://registry.svc.ci.openshift.org/)?

Author: Added default values as suggested.
